build.gradle

```diff
@@ -53,26 +56,52 @@ dependencies {
     testImplementation 'junit:junit:4.12'
 }
 
-// create a release zip
-task release(type: Zip) {
+def readModInfoJson() {
     def jsonFile = file('src/main/resources/mod_info.json')
-    def slurper = new groovy.json.JsonSlurper()
-    def parsedJson = slurper.parse(jsonFile)
+    def slurper = new JsonSlurper()
+    return slurper.parse(jsonFile)
+}
+
+task release(type: Zip) {
     dependsOn build
-    archiveFileName = "${project.name}-${parsedJson.version}.zip"
+    archiveFileName = "${project.name}-${readModInfoJson().version}.zip"
     destinationDirectory = file(rootDir)
     from sourceSets.main.resources
     from jar
     into project.name
 }
 
-task copyToMods(type: Copy) {
-    if (!project.hasProperty('game.full_path')) {
-        println "WARNING: Missing game.full_path in gradle.properties. See gradle.properties.dist as an example."
-        return;
+def arePropertiesDefined() {
+    if (!project.hasProperty('game.dir') || !project.hasProperty('game.mod_folder')) {
+        println 'WARNING: Missing properties in gradle.properties. See gradle.properties.dist as an example.'
+        println 'WARNING: Task execution stopped.'
+        return false
     }
+    return true
+}
+
+def getModDir() {
+    return project.getProperty('game.dir') + project.getProperty('game.mod_folder') + readModInfoJson().id
+}
 
+task copyToMods(type: Copy) {
+    if (!arePropertiesDefined()) {
+        return
+    }
     dependsOn build
     from jar
-    into project.getProperty('game.full_path')
-}
\ No newline at end of file
+    into getModDir()
+}
+
+def getCoreDirForWindowsOrMainPathForLinux() {
+    return Os.isFamily(Os.FAMILY_WINDOWS) ? project.getProperty('game.dir') + '/starsector-core/' : project.getProperty('game.dir')
```

Comment:
This is inconsistent. You expect Windows path to have a trailing slash, while Linux not to have one.